doc/design/model_guide.md

```diff
@@ -0,0 +1,36 @@
+# SQLFlow Models User Guide
+
+There are various models in SQLFlow Model Zoo, and as the number of SQLFlow users increases,
+an important question for users raised -- **How can I choice a model and how can I use it**.
+This design provides an outline of the model user guide to help users choose a model and use it in his/her business scene.
+
+## Model Categories
+
+In the first section, we list all the model categories, e.g., Classification, Regression, Unsupervised Clustering,
+and time-series. Each type leaves a summary. Each category mapping to one or more Model implementation,
```

Comment:
Time series models are regression models. Maybe we need more real case facing categories like:

- risk models ( credit card fraud predicting, score card etc.)
- time series models
- ranking ( CTR models etc.)
- user preferences

Reply:
Need "steps to select the model" guide like:

Your case -> case category -> models for the case -> how to select models in your scenario -> test and compare models -> finish

Reply:
> risk models ( credit card fraud predicting, scorecard etc.)

credit card fraud predicting is not a model instead of a tutorial with DNNClassifier model on the credit card dataset.

Reply:
I updated this document referenced the algorithm category on https://docs.microsoft.com/en-us/azure/machine-learning/algorithm-cheat-sheet?WT.mc_id=docs-article-lazzeri , which designs a model category on users' purpose on the data, I think it makes sense.

Also, I think we should list the business scenario e.g. risk models, ranking .e.g. but it's better to have one or more tutorials on theses scenario, how do you think?

Reply:
Maybe this kind of "best practice of how to select a model" is straight forward yet it's aiming to help users that already know what should he/she do.

If the user is not quite sure what to do when the business case arrives, he/she can not make use of this list.

Reply:
> If the user is not quite sure what to do when the business case arrives, he/she can not make use of this list.

he/she can check the exciting business cases(tutorials) and find a similar case on their data set, so we need two lists?
1. If you know what to do with your data, you can check this list ...
1. If you don't know what to do, you can check the exciting business cases list ...

Reply:
We can provide some solution templates for various scenarios. It contains date preprocessing and various models (PK/compare with each other) in this template.